agent/engine/docker_task_engine.go

```diff
@@ -625,7 +650,22 @@ func (engine *DockerTaskEngine) startContainer(task *api.Task, container *api.Co
 			Error: CannotStartContainerError{fmt.Errorf("Container not recorded as created")},
 		}
 	}
-	return client.StartContainer(dockerContainer.DockerID, startContainerTimeout)
+	md := client.StartContainer(dockerContainer.DockerID, startContainerTimeout)
+
+	//Get metadata through container inspection and available task information then convert this to writable metadata 
+	rawContainer, ioerr := client.InspectContainer(dockerContainer.DockerID, inspectContainerTimeout)
+	if ioerr != nil {
```

Comment:
can you change `ioerr` to `err`, multiple places. Also can you extract this part as a function?